### PR TITLE
Jgfouca/upstream merge 2024 11 14

### PR DESCRIPTION
**Pull request by jgfouca**
@dqwu Asked for this. This should be one of the last upstream merges.

### CODE REVIEW

File: components/eamxx/CMakeLists.txt [hunk 1]
```diff
@@ -200,6 +200,9 @@ option(SCREAM_MPI_ON_DEVICE "Whether to use device pointers for MPI calls" ON)
 option(SCREAM_ENABLE_MAM "Whether to enable MAM aerosol support" ON)
 set(SCREAM_SMALL_KERNELS ${DEFAULT_SMALL_KERNELS} CACHE STRING "Use small, non-monolothic kokkos kernels for ALL components that support them")
 set(SCREAM_P3_SMALL_KERNELS ${SCREAM_SMALL_KERNELS} CACHE STRING "Use small, non-monolothic kokkos kernels for P3 only")
+message("JGF DEFAUL_SMALL_KERNELS is ${DEFAULT_SMALL_KERNELS}")
```
Comment (bartgol) [inline components/eamxx/CMakeLists.txt:203]: Was this meant to be just some debugging stuff?

**Review by bartgol — APPROVED** [review]

File: components/eamxx/CMakeLists.txt (hunk 1 above)
Comment (jgfouca) [reply components/eamxx/CMakeLists.txt:203]: Yes, whoops! Let me remove that

**Review by bartgol — APPROVED** [review]

**Comment by bartgol** [comment]
Noting that the automerge feature of mergify worked (after not working for a long time). The merge commit is formatted with the mergify specs:

```
    Merge pull request #3115 from jgfouca/upstream_merge_2024_11_14
    
    Automatically merged using mergify
    PR title: Jgfouca/upstream merge 2024 11 14
    PR author: jgfouca
    PR labels: ['CI: workflow change approved', 'CI: automerge']
```

**Comment by bartgol** [comment]
> @bartgol why did the tests not run tho?

That is a very good question. The output of pre_process_pr shows "No relevant files touched by this PR", which is obviously not true, given the changes in components/eamxx. Perhaps I need to crank up the stuff I print in that action (e.g., the list of changed files, the patterns, the matches), so we can debug when stuff like this happens.

**Pull request merged**